src/build/optimize-streams.ts

```diff
@@ -0,0 +1,110 @@
+import {minify as uglify, MinifyOptions as UglifyOptions} from 'uglify-js';
+import {css as cssSlam} from 'css-slam';
+import {minify as htmlMinify, Options as HTMLMinifierOptions} from 'html-minifier';
+import {Transform} from 'stream';
+import * as logging from 'plylog';
+import File = require('vinyl');
+
+let logger = logging.getLogger('cli.build.optimize-streams');
+
+export type FileCB = (error?: any, file?: File) => void;
+export type CSSOptimizeOptions = {
+  stripWhitespace?: boolean;
+}
+
+/**
+ * GenericOptimizeStream is a generic optimization stream. It can be extended
+ * to create a new kind of specific file-type optimizer, or it can be used
+ * directly to create an ad-hoc optimization stream for certain types of files.
+ */
+export class GenericOptimizeStream extends Transform {
+
+  validExtension: string;
+  optimizer: (content: string, options: any) => string;
+  optimizerOptions: any;
+
+  constructor(
+      validExtension: string,
+      optimizer: (content: string, optimizerOptions: any) => string,
+      optimizerOptions: any) {
+    super({objectMode: true});
+    this.optimizer = optimizer;
+    this.validExtension = validExtension;
+    this.optimizerOptions = optimizerOptions || {};
+  }
+
+  _transform(file: File, encoding: string, callback: FileCB): void {
+    if (file.contents && file.path.endsWith(`${this.validExtension}`)) {
+      try {
+        let contents = file.contents.toString();
+        contents = this.optimizer(contents, this.optimizerOptions);
+        file.contents = new Buffer(contents);
+      } catch (err) {
```

Comment:
Good to know, I've always used `err`.  The library  is using `err`, `error`, and even `e` all over the place, I'll change this here and then create a separate PR to get the rest of the repo on board.